### DISCUSSION TITLE
PyScript - "Next" 2023.09.1.RC1

### DISCUSSION THREAD

**Discussion by tedpatrick** [pyscript/pyscript#1744] [Announcements]
# PyScript - "Next" 2023.09.1.RC1 🎉

After a huge amount of effort, we are delighted to announce the release of the first technical preview of the next version of PyScript.

https://pyscript.net/unstable/
https://pyscript.net/snapshots/2023.09.1.RC1/

This is a big change 🔀 This is a work in progress 📈 This is where we'd love your feedback 💬

In summary:

* We have re-architected PyScript on top of a new core: [Polyscript](https://github.com/pyscript/polyscript).
* This has significantly reduced the size of PyScript and improved its performance.
* PyScript supports multiple Python interpreters ([Pyodide](https://pyodide.org/) and [MicroPython](https://micropython.org/)).
* PyScript now supports parallel execution of code via web workers.
* There is a new plugin system.
* We've started a complete re-write of [the documentation](https://docs.pyscript.net).

Changelog:
* We re-wrote everything. 🚀 

Remember, this is a technical preview! 🤖

There will be bugs 🐛 Things may not work as they used to 💥 Features will be missing 🔍

We are in a state of transition as we add missing features, fix bugs and update things as we listen to your feedback. Help improve our work via [our GitHub project](https://github.com/pyscript) or join in the discussion via [our discord server](https://discord.gg/HxvBtukrg2).

The second technical preview will arrive very soon.

Onwards! 🚀

**Comment by shivaylamba** [comment]
This is amazing!!!!!!

**Comment by stonethedev** [comment]
Great job, keep at it

**Comment by jd-solanki** [comment]
Awesome ❤️

**Comment by antocuni** [comment]
This new release is a very big milestone for PyScript, because it unlocks *a lot* of potential.
I know that everybody in the team has their own favorite feature of PyScript Next: some like the MicroPython support, some the smaller download size, some the faster startup.

Let me share with the community what *I* think it is the most exciting feature: PyScript next allows to transparently run Python code inside a webworker, and thanks to the magic wonders of `xworker.sync`, it allows to fill the gap between the async-first nature of the browser and the sync-first nature of the vast majority of Python code around.

In PyScript classic, it was technically impossible to support or even emulate a lot of basic blocking APIs which Python programmers take for granted, such as:
- `input()` for reading a string from the keyboard
- any kind of synchronous socket such as `requests.get()` or `sock.recv()`
- any kind of support for debuggers (including, but not limited to, `pdb.set_trace()`)

With Next, all of the above is possible, when you run in a worker. The following are a couple of demos which demonstrate the potential:
- https://antocuni.pyscriptapps.com/pyterminal-poc/v1/: this revives `py-terminal` by using xterm.js and a blocking version of `input()` based on `xworker.sync`: the result is that you can for example run pdb (or even pdb++, like in this example)
- https://antocuni.pyscriptapps.com/pytest-in-pyterminal/v1/: similar to the above, but shows how to run an entire pytest session inside a terminal
- https://antocuni.pyscriptapps.com/gui-debugger/v1/: this is a very raw proof of concept, but shows how to connect a GUI debugger running in the main thread with a user program running in a worker. The GUI is very basic and ugly, but you can imagine having one with all the bells&whistles that normal debuggers have.

The plan is to re-integrate the py-terminal into pyscript in the next weeks.
